DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift

```diff
@@ -237,6 +270,7 @@ final class SubscriptionPagesUseSubscriptionFeature: Subfeature, ObservableObjec
     func featureSelected(params: Any, original: WKScriptMessage) async throws -> Encodable? {
```

Comment:
(general comment, not only for this function)

Overall the pattern of having functions that express a failure returning a nil value is very Objective-C, the return value should be optional only if an optional value makes sense for a positive outcome in the context and not as a way to signal an error, the errors should be propagated by throwing exceptions. 
Additionally, modifying an external variable for storing errors makes the function's interface obscure.

Suggestions:
- make the return value not optional
- propagate errors throwing them and ctaching them in the function callers

Reply:
Agree on the nil value return type.  

As for propagating errors via exceptions, that's the ideal scenario, sure; but in this context of the UserScripts architecture, class methods are invoked by external sources (webviews > userscripts), and view models do not directly control when these methods are executed.   Propagating errors and other state changes through published properties is a necessity rather than a choice.

Reply:
Now that I recall, we're returning an optional, because this method is called from the `UserScriptMessageBroker`, which expects an optional message.     While I guess this could be updated, the scope of the subscription project did not include the development of these frameworks.-